### PR TITLE
Update CMakeLists.txt [Optional Installation]

### PR DESCRIPTION
**Pull request by kenkit**
Optional Install.
This was was adding header files to my msi package

### CODE REVIEW

**Comment by vasild** [comment]
Hello,

Isn't not running `make install` sufficient? Why do you need to disable the `install()` CMake instructions?

https://cmake.org/cmake/help/latest/command/install.html
> Rules specified by calls to this command within a source directory are executed in order **during installation**

Thanks!

**Comment by kenkit** [comment]
Sorry I couldn't respond sooner. 
The install option is ambiguous in windows platform since it ends up adding the headers to cpack installation.
I found that most people make it optional or disabled on windows platform.

**Comment by vasild** [comment]
> The install option is ambiguous in windows platform since it ends up adding the headers to cpack installation.

How do you install on Windows? Do you run [cmake --install](https://cmake.org/cmake/help/latest/manual/cmake.1.html#install-a-project)? What is puzzling me is that you disabled all [install](https://cmake.org/cmake/help/latest/command/install.html) instructions and that is equivalent to just not running the install step, i.e. not installing, i.e. not running `make install`, `cmake --install` or whatever you do on Windows.

> EDIT:I didn't disable it, I just made it optional

By default it gets disabled. I would merge this change, even though I do not understand the mechanics of installation on Windows (you help me understand that!). But can you revert the logic so that if you do nothing then, by default, the install works as before. And only if you define a variable (in the environment or on the `cmake` command line), then the install is skipped.

PS for consistency with the rest of the `CMakeLists.txt` use lowercase `if()` and `endif()`.

Thanks!

**Comment by kenkit** [comment]
No. I use this package as a dependency in one of my projects.
My projects uses add_subdirectory in cmake and also uses the install option to install my projects files.
Since cpp-ipfs also calls install cmake will assume that the files in cpp-ipfs should also be installed.
Atleast two other projects I use have made the install optional e.g

https://github.com/libarchive/libarchive/blob/2f3033ca23f8c21160506c3c7ac8a0df0d3fde42/CMakeLists.txt#L224

https://github.com/rikyoz/bit7z/pull/27#issuecomment-533498870

**Comment by kenkit** [comment]
Ok let me update the CMakeLists.txt

File: CMakeLists.txt [hunk 1]
```diff
@@ -44,11 +44,11 @@ set_target_properties(${IPFS_API_LIBNAME} PROPERTIES
   VERSION ${PROJECT_VERSION}
 )
 target_link_libraries(${IPFS_API_LIBNAME} ${CURL_LIBRARIES})
-
-install(TARGETS ${IPFS_API_LIBNAME} DESTINATION lib)
-install(FILES include/ipfs/client.h DESTINATION include/ipfs)
-install(FILES include/ipfs/http/transport.h DESTINATION include/ipfs/http)
-
+if(ENABLE_INSTALL)
+   install(TARGETS ${IPFS_API_LIBNAME} DESTINATION lib)
```
Comment (vasild) [inline CMakeLists.txt]: 3 spaces as indentation whereas the rest of the file uses 2.

File: CMakeLists.txt [hunk 2]
```diff
@@ -44,11 +44,11 @@ set_target_properties(${IPFS_API_LIBNAME} PROPERTIES
   VERSION ${PROJECT_VERSION}
 )
 target_link_libraries(${IPFS_API_LIBNAME} ${CURL_LIBRARIES})
-
-install(TARGETS ${IPFS_API_LIBNAME} DESTINATION lib)
-install(FILES include/ipfs/client.h DESTINATION include/ipfs)
-install(FILES include/ipfs/http/transport.h DESTINATION include/ipfs/http)
-
+if(ENABLE_INSTALL)
```
Comment (vasild) [inline CMakeLists.txt]: That is still disabling the install by default. What about `if(NOT DISABLE_INSTALL)`? Then the default behavior is unchanged and you need to explicitly define `DISABLE_INSTALL` to skip installing.

Comment (kenkit) [reply CMakeLists.txt]: Let me fix it.
Thanks

File: CMakeLists.txt (hunk 1 above)
Comment (kenkit) [reply CMakeLists.txt]: Fixed identation

**Comment by kenkit** [comment]
Hi the failed test is not as a result of my changes.
All other tests seem to be fine.

**Comment by vasild** [comment]
Merged, thanks!

**Pull request merged**